### PR TITLE
feat: migrate policy APIs to Connect RPC

### PR DESCRIPTION
**Pull request by rohilsurana**
## Summary
Migrate 6 policy APIs from gRPC to Connect RPC format as part of the IAM API standardization initiative.

## APIs to Migrate
- [x] CreatePolicy - ✅ **COMPLETED** (commit: 23aa904f)
- [x] GetPolicy - ✅ **COMPLETED** (commit: 7d19c25d)
- [x] ListPolicies - ✅ **COMPLETED** (commit: c7212419)
- [x] UpdatePolicy - ✅ **COMPLETED** (commit: 7067de7e)
- [x] DeletePolicy - ✅ **COMPLETED** (commit: 62ae3f27)
- [x] CreatePolicyForProject - ✅ **COMPLETED** (commit: 23ad5d1f)

### ✅ CreatePolicy (COMPLETED)
**Files Changed:**
- `internal/api/v1beta1connect/policy.go` - Connect RPC implementation
- `internal/api/v1beta1connect/policy_test.go` - Comprehensive test suite with 8 scenarios
- `internal/api/v1beta1connect/mocks/policy_service.go` - Mock service interface

**Key Features:**
- ✅ Connect RPC request/response wrapping with `connect.Request[T]` and `connect.Response[T]`
- ✅ Proper error handling with Connect error codes (InvalidArgument, Internal)
- ✅ Namespace splitting for resource and principal validation
- ✅ Metadata handling for policy attributes
- ✅ Audit logging for policy creation events
- ✅ Identical business logic preservation from gRPC version
- ✅ 8 comprehensive test scenarios covering all edge cases

**Test Coverage:**
- Invalid resource/principal namespace splitting errors
- Role ID validation errors
- Policy detail validation errors
- Service layer error handling
- Successful policy creation (basic data)
- Successful policy creation (with metadata)
- Group principal support
- Metadata transformation errors

### ✅ GetPolicy (COMPLETED)
**Files Changed:**
- `internal/api/v1beta1connect/errors.go` - Added `ErrPolicyNotFound` error constant
- `internal/api/v1beta1connect/policy.go` - Added GetPolicy Connect RPC implementation
- `internal/api/v1beta1connect/policy_test.go` - Added comprehensive test suite

**Key Features:**
- ✅ Connect RPC request/response wrapping
- ✅ Proper error mapping for policy retrieval scenarios
- ✅ Connect error codes: `CodeNotFound` for missing policies, `CodeInternal` for service errors
- ✅ Policy transformation with metadata support
- ✅ Identical business logic preservation from gRPC version
- ✅ 6 comprehensive test scenarios covering all edge cases

**Test Coverage:**
- Internal server error when policy service fails
- Not found error when ID is empty (ErrInvalidID)
- Not found error when ID is not UUID (ErrInvalidUUID)
- Not found error when policy doesn't exist (ErrNotExist)
- Successful policy retrieval with all fields
- Metadata transformation error handling

### ✅ ListPolicies (COMPLETED)
**Files Changed:**
- `internal/api/v1beta1connect/policy.go` - Added ListPolicies Connect RPC implementation and resolveFilter helper
- `internal/api/v1beta1connect/policy_test.go` - Added comprehensive test suite

**Key Features:**
- ✅ Connect RPC request/response wrapping
- ✅ Policy filtering with entity resolution for non-UUID filters (org, role, project, user)
- ✅ Connect error codes: `CodeInvalidArgument` for filter errors, `CodeInternal` for service errors
- ✅ Policy transformation with metadata support
- ✅ Identical business logic preservation from gRPC version including resolveFilter functionality
- ✅ 4 comprehensive test scenarios covering all edge cases

**Test Coverage:**
- Internal server error when policy service fails
- Successful policy listing with multiple policies
- Empty results handling (nil slice behavior)
- Metadata transformation error handling

### ✅ UpdatePolicy (COMPLETED)
**Files Changed:**
- `internal/api/v1beta1connect/policy.go` - Added UpdatePolicy Connect RPC implementation
- `internal/api/v1beta1connect/policy_test.go` - Added test suite

**Key Features:**
- ✅ Connect RPC request/response wrapping
- ✅ Unimplemented behavior matching original gRPC version
- ✅ Connect error code: `CodeUnimplemented` with proper error message
- ✅ Identical unimplemented status preservation from gRPC version
- ✅ 1 test scenario verifying unimplemented behavior

**Test Coverage:**
- Unimplemented error response validation
- Correct Connect error code verification
- Error message format confirmation

### ✅ DeletePolicy (COMPLETED)
**Files Changed:**
- `internal/api/v1beta1connect/policy.go` - Added DeletePolicy Connect RPC implementation
- `internal/api/v1beta1connect/policy_test.go` - Added comprehensive test suite

**Key Features:**
- ✅ Connect RPC request/response wrapping
- ✅ Comprehensive error handling with proper Connect error codes
- ✅ Connect error codes: `CodeNotFound` for missing policies, `CodeInvalidArgument` for invalid details, `CodeAlreadyExists` for conflicts, `CodeInternal` for service errors
- ✅ Audit logging preservation for successful deletions
- ✅ Identical business logic preservation from gRPC version
- ✅ 8 comprehensive test scenarios covering all edge cases

**Test Coverage:**
- Policy not found scenarios (3 test cases: ErrNotExist, ErrInvalidID, ErrInvalidUUID)
- Invalid argument scenarios (2 test cases: ErrInvalidDetail, namespace.ErrNotExist)
- Conflict scenario (1 test case: ErrConflict)
- Internal server error scenario (1 test case: unknown errors)
- Successful deletion scenario (1 test case: with audit logging)

### ✅ CreatePolicyForProject (COMPLETED)
**Files Changed:**
- `internal/api/v1beta1connect/errors.go` - Added `ErrProjectNotFound` error constant
- `internal/api/v1beta1connect/policy.go` - Added CreatePolicyForProject Connect RPC implementation
- `internal/api/v1beta1connect/policy_test.go` - Added comprehensive test suite

**Key Features:**
- ✅ Connect RPC request/response wrapping with proper proto message types
- ✅ Request body validation (role ID and principal required)
- ✅ Principal namespace splitting and validation
- ✅ Project lookup and validation with proper error handling
- ✅ Connect error codes: `CodeInvalidArgument` for validation errors, `CodeNotFound` for missing projects, `CodeInternal` for service errors
- ✅ Policy creation with project as resource scope
- ✅ Audit logging preservation for successful creations
- ✅ Identical business logic preservation from gRPC version
- ✅ 9 comprehensive test scenarios covering all edge cases

**Test Coverage:**
- Invalid argument errors (nil body, empty role ID, empty principal)
- Principal namespace splitting validation
- Project not found scenario
- Role ID validation errors
- Policy detail validation errors
- Internal server errors
- Successful policy creation for project

## Testing
- ✅ All existing tests pass
- ✅ New Connect RPC tests added with comprehensive coverage
- ✅ Build verification completed
- ✅ Lint checks passed

## Migration Pattern
Each API follows the established Connect RPC migration pattern:
1. Request/response type wrapping
2. Access request data via `request.Msg.GetFieldName()`
3. Wrap responses with `connect.NewResponse()`
4. Convert errors to Connect error format
5. Preserve identical business logic
6. Comprehensive test migration

## Status
✅ **MIGRATION COMPLETE** - All 6 policy APIs successfully migrated to Connect RPC!

### CODE REVIEW

File: internal/api/v1beta1connect/policy.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, ErrInternalServerError)
+	}
+
+	return connect.NewResponse(&frontierv1beta1.GetPolicyResponse{Policy: policyPB}), nil
+}
+
+func (h *ConnectHandler) UpdatePolicy(ctx context.Context, request *connect.Request[frontierv1beta1.UpdatePolicyRequest]) (*connect.Response[frontierv1beta1.UpdatePolicyResponse], error) {
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("unsupported at the moment"))
+}
```
Comment (whoAbhishekSah) [inline internal/api/v1beta1connect/policy.go]: We can skip this as the ConnectHandler struct will return an Unimpl error by default for this RPC.
